### PR TITLE
first version of YAML parser

### PR DESCRIPTION
**Pull request by elphastori**

### CODE REVIEW

File: parser/yaml/yaml.go [hunk 1]
```diff
@@ -0,0 +1,831 @@
+//  Copyright 2013 Walter Schulze
```
Comment (awalterschulze) [inline parser/yaml/yaml.go:1]: You can put your name and the year here :)

**Review by awalterschulze — COMMENTED** [review]
Thanks for starting this work.
I always start by reviewing the tests before reviewing the code, since it is more important what the code does that what it looks like.
From a quick skim though this seems to be on the right track :)

File: parser/yaml/yaml.go (hunk 1 above)
Comment (elphastori) [reply parser/yaml/yaml.go]: Thanks!

Comment (elphastori) [reply parser/yaml/yaml.go]: Thanks for the review. I've added unit tests that cover the most common cases based on the XML and JSON parsers. I'll definitely add a lot more as I make the YAML implementation more robust.

File: README.md [hunk 2]
```diff
@@ -9,7 +9,15 @@
 Currently there are three tools in the katydid toolkit:
 
   * Relapse: a regular expression type language for trees that matches up to 1000000s of records per second, 
-  * A collection of parsers (protobuf, json, xml, reflected go structures) which are easily extendable, and
+  * A collection of parsers (protobuf, json, xml, reflected go structures, yaml*) which are easily extendable, and
   * A collection of encoders (protobuf, json, xml, reflected go structures) which are useful for dynamic transcoding.
 
 [more...](http://katydid.github.io)
+
+*The YAML parser implementation is in progress and has the following knwown limitations
```
Comment (awalterschulze) [inline README.md:17]: Maybe let's create a .md file in the yaml folder and add this there or even as a comment in the godoc of the yaml.go file, what do you think?

File: parser/yaml/yaml.go [hunk 3]
```diff
@@ -0,0 +1,831 @@
...
+
+	if line.hasValue { // tighten this condition
+		value := []byte(strings.TrimSpace(string(line.value)))
+		if len(value) > 0 {
+			return s.scanString()
+		}
+	}
+
+	return s.scanString()
+	// return s.expected("objects not supported")
```
Comment (awalterschulze) [inline parser/yaml/yaml.go:90]: maybe we can remove the comment? and a few others below, or do you want to maybe expand on them later, if so let's add that to the comment.

**Review by awalterschulze — COMMENTED** [review]
Small nits, but other wise looks good to merge :D

File: README.md (hunk 2 above)
Comment (elphastori) [reply README.md:17]: I agree, this leaves the readme clean. I'll go with the .md file in the yaml folder.

File: parser/yaml/yaml.go (hunk 3 above)
Comment (elphastori) [reply parser/yaml/yaml.go:90]: Removed redundant comments in YAML parser implementation

**Comment by elphastori** [comment]
Ran `$ make travis` for testing. Use version 1.0.0 of github.com/kisielk/errcheck because master branch no longer supports Go 1.10.x

**Review by awalterschulze — APPROVED** [review]
Great code, thank you so much for starting this effort.

**Pull request merged**